<launch>
  <!-- Arguments -->

  <arg name="cmd_vel_topic" default="/mechROS/base_control/cmd_vel" />
  <arg name="odom_topic" default="/odometry_rel/filtered" />
  <arg name="base_global_planner" default="global_planner/GlobalPlanner"/>
  <!-- <arg name="base_local_planner" default="eband_local_planner/EBandPlannerROS"/> -->
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>



  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="Navigator" output="screen">

    <param name="base_local_planner" value="$(arg base_local_planner)" />
    <param name="base_global_planner" value="$(arg base_global_planner)" />

    <rosparam file="$(find mech_ros)/navigation/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mech_ros)/navigation/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mech_ros)/navigation/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find mech_ros)/navigation/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find mech_ros)/navigation/move_base_params.yaml" command="load" />
    <rosparam file="$(find mech_ros)/navigation/global_planner_params.yaml" command="load" />
    <rosparam file="$(find mech_ros)/navigation/dwa_local_planner_params.yaml" command="load" />
    
    <!-- <rosparam file="$(find mech_ros)/navigation/eband_local_params.yaml" command="load" /> -->
    <remap from="/cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>

  </node>

</launch>
